src/DefinitionResolver.php

```diff
@@ -144,6 +150,10 @@ public function resolveReferenceNodeToDefinition(Node $node)
     {
         // Variables are not indexed globally, as they stay in the file scope anyway
         if ($node instanceof Node\Expr\Variable) {
+            // Resolve $this
+            if ($node->name === 'this' && $fqn = $this->getContainingClassFqn($node)) {
+                return $this->index->getDefinition($fqn, false);
+            }
```

Comment:
It is actually needed in order to provide signatureHelp when invoking a method on `$this`, and this is not part of the $this autocomplete PR, as it was not needed for autocomplete - different parts of the code are involved. I could move it to a separate PR?